cf/read_write/read.py

```diff
@@ -930,34 +914,17 @@ def _read_a_file(
     umversion = 405
 
     if um:
-        # ftype = 'UM'
         fmt = um.get("fmt")
         word_size = um.get("word_size")
         endian = um.get("endian")
-        umversion = um.get("version")
+        umversion = um.get("version", umversion)
         height_at_top_of_model = um.get("height_at_top_of_model")
-        if fmt in ("PP", "pp", "pP", "Pp"):
-            fmt = fmt.upper()
-            # For PP format, there is a default word size and
-            # endian-ness
-            if word_size is None:
-                word_size = 4
 
-            if endian is None:
-                endian = "big"
```

Comment:
Aren't these important, still? Not sure why you've cut them given the nature of this PR, so just to check there's a reason?

Reply:
Good question. Forcing `word_size` and `endian` to have defaults for PP was basically wrong, and the code now behaves as intended - we see as many little as big endian PP files (even though the PP standard mandates big endian). Simmilary for word size. Nearly all PP fiels are 32-bit, but there's always one ...

Reply:
I see, thanks for clarifying. Interesting context!